service/history/replication/executable_task_tracker.go

```diff
@@ -130,30 +131,38 @@ func (t *ExecutableTaskTrackerImpl) LowWatermark() *WatermarkInfo {
 	t.Lock()
 	defer t.Unlock()
 
+	element := t.taskQueue.Front()
 Loop:
-	for element := t.taskQueue.Front(); element != nil; element = element.Next() {
+	for element != nil {
 		task := element.Value.(TrackableExecutableTask)
 		taskState := task.State()
 		switch taskState {
 		case ctasks.TaskStateAcked:
+			nextElement := element.Next()
 			delete(t.taskIDs, task.TaskID())
 			t.taskQueue.Remove(element)
+			element = nextElement
```

Comment:
this is the real bugfix, i.e. need to keep track of the next element when deleting the current element